### PR TITLE
Fix Reference Object usage on parameters for swagger spec

### PR DESCRIPTION
**Pull request by RafaeLeal**
As I commented in the issue
https://github.com/oliyh/martian/issues/195

This is not working for swagger, but it is for OpenAPI v3. I tried to refactor the code to use the same logic, adding tests for it.

### CODE REVIEW

File: core/test/martian/swagger_test.cljc [hunk 1]
```diff
@@ -18,6 +18,28 @@
     (is (= {:id s/Str} (:path-schema get-handler)))
     (is (= nil (:query-schema get-handler)))))
 
+(deftest parameter-schemas-ref-test
```
Comment (RafaeLeal) [inline core/test/martian/swagger_test.cljc:21]: This is the main scenario of the bug

File: core/src/martian/schema.cljc [hunk 2]
```diff
@@ -71,9 +72,31 @@
...
+(defn lookup-ref
+  [ref ref-lookup]
+  (if (string/starts-with? ref "#/")
+    (or (get-in ref-lookup (drop 1 (map keyword (string/split ref #"/"))))
+        (throw (ex-info "Cannot find reference"
+                        {:reference ref})))
+    (throw (ex-info "Non-local references are not supported yet. References should start ref with '#/'"
+                    {:reference ref}))))
+
+(defn resolve-ref-object
```
Comment (RafaeLeal) [inline core/src/martian/schema.cljc:84]: this function follows through when you have `$ref` pointing to other `$ref`. This is possible following OpenAPI v3, but not Swagger. I believe this scenario was not covered previously, so I added it here.

File: core/test/martian/schema_test.cljc [hunk 3]
```diff
@@ -400,3 +401,34 @@
...
     (is (= (s/maybe {(s/optional-key :b) (s/maybe {(s/optional-key :a) s/Any})})
            schema))))
+
+
+
+(deftest resolve-ref-object-test
+  (let [params       {:idParam {:name "id" :in "query"}}
+        swagger-spec {:definitions params}
+        openapi-spec {:components {:parameters params}}
+        #?(:cljs ExceptionInfo) #?(:cljs js/Error)]
```
Comment (RafaeLeal) [inline core/test/martian/schema_test.cljc:411]: This is a hack to work with ClojureScript..
we might want to change it to some test-helper like this:
https://gist.github.com/metametadata/69cae3488d891f5e23d9

Comment (oliyh) [reply core/test/martian/schema_test.cljc:411]: I tend to just do this at the top level:
```
#?(:cljs (def ExceptionInfo js/Error))
```

**Comment by oliyh** [comment]
This is great, thank you!

**Comment by RafaeLeal** [comment]
Thanks for the merge!
Can you generate a version for it? 
I'm using it on [k8s-api](https://github.com/nubank/k8s-api) and the newer versions of Kubernetes needs it

**Comment by oliyh** [comment]
Hi @RafaeLeal I have release `0.1.26`

**Comment by RafaeLeal** [comment]
@oliyh Thanks!

**Pull request merged**